src/Compilers/CSharp/Portable/BoundTree/BoundExpressionExtensions.cs

```diff
@@ -42,6 +44,45 @@ public static RefKind GetRefKind(this BoundExpression node)
                 case BoundKind.ImplicitIndexerAccess:
                     return ((BoundImplicitIndexerAccess)node).IndexerOrSliceAccess.GetRefKind();
 
+                case BoundKind.ConditionalOperator:
+                    {
+                        var conditionalOperator = (BoundConditionalOperator)node;
+                        if (!conditionalOperator.IsRef)
+                        {
+                            return RefKind.None;
+                        }
+                        {
+                            if (tryGetReadOnlyRefKind(conditionalOperator.Consequence) is { } r)
```

Comment:
>tryGetReadOnlyRefKind [](http://example.com/codeflow?start=32&length=21)

The intent should be clearly captured in comment explaining why is this necessary. Also, the need to do this should be backed by tests, their behavior should observably change if the logic around "readonly" is "broken" or removed.